explorer/client/src/components/ownedobjects/OwnedObjects.tsx

```diff
@@ -42,24 +47,67 @@ function OwnedObjectStatic({ objects }: { objects: string[] }) {
     return <OwnedObjectView results={results} />;
 }
 
+type IdResultsPair = [string, GetObjectInfoResponse];
+
 function OwnedObjectAPI({ objects }: { objects: string[] }) {
     const [results, setResults] = useState(DATATYPE_DEFAULT);
     const [isLoaded, setIsLoaded] = useState(false);
 
     useEffect(() => {
-        Promise.all(objects.map((objID) => rpc.getObjectInfo(objID))).then(
-            (results) => {
-                setResults(
-                    results.map(({ id, objType, version, data }) => ({
-                        id: id,
-                        Type: objType,
-                        Version: version,
-                        display: processDisplayValue(data.contents?.display),
-                    }))
-                );
-                setIsLoaded(true);
-            }
-        );
+        Promise.all(
+            objects.map(
+                (objID) =>
+                    new Promise<IdResultsPair>((resolve, reject) => {
+                        rpc.getObjectInfo(objID)
+                            .then((res) => {
+                                resolve([objID, res]);
+                            })
+                            .catch((err) => {
+                                reject(err);
+                            });
+                    })
+            )
+        ).then((pairs) => {
+            const results = pairs.map((pair) => {
+                const result = pair[1];
+
+                if (
+                    result.status === 'Exists' &&
+                    isObjectExistsInfo(result.details)
+                ) {
+                    const obj = result.details.object;
+                    return {
+                        id: pair[0],
+                        Type: obj['objType'] as string,
+                        Version: (obj['version'] as number).toString(),
+                        display: processDisplayValue(obj['display'] as string),
+                    };
+                }
+                if (
+                    result.status === 'Deleted' &&
+                    isObjectRef(result.details)
+                ) {
+                    return {
```

Comment:
nit: I would add a status field to signify the object is deleted or not exists, so that the front end can render them differently.